forte/data/multi_pack.py

```diff
@@ -176,6 +176,172 @@ def get_span_text(self, begin: int, end: int):
             "specific data pack to get text."
         )
 
+    def remove_pack(
+        self, index_of_pack: int, clean_invalid_entries: bool = False
+    ) -> bool:
+        """
+        Remove a data pack at index `index_of_pack` from this multi pack.
+
+        In a multi pack, the data pack to be removed may be associated with
+        some multi pack entries, such as `MultiPackLinks` that are connected
+        with other packs. These entries will become dangling and invalid,
+        thus need to be removed. One can consider removing these links before
+        calling this function, or set the `clean_invalid_entries` to `True` so
+        that they will be automatically pruned.
+
+        Args:
+            index_of_pack (int): The index of pack for removal from the
+              multi pack. If invalid, no pack will be deleted.
+            clean_invalid_entries (bool): Switch for automatically cleaning
+              the entries associated with the data pack being deleted which
+              will become invalid after the removal of the pack. Default is
+              False.
```

Comment:
Could we make `purge_deleted_packs` callable by setting a bool here?

Reply:
I think allowing easy "auto purge" (change index) might easily cause runtime error for user if user does not understand the whole process of pack removal and implication/consequences to indexes of packs after removal very clearly -- this is  not very obvious to user.  Because  Basically there are 2 different strategies for removing pack from a multi-pack: 
1) index-safe removal. 2) None-gap removal (auto purge).  So instead of adding a parameter which user could easily select to "purge" without thinking much and believe all things can be automatically handled, which could cause dynamic and subtle (not apparent) errors when user try to remove an other pack if failed to account for index change caused by strategy 2. So a better way to remind user could be making the method name apparent and according to different strategies so that user must know he/she is intended to do before trying to remove a pack -- we can all one method index_safe_remove and the other called index_changing_remove (or no_gap_remove)